hackathon_site/dashboard/frontend/src/components/general/ProductOverview/ProductOverview.test.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import SideSheetRight from "../SideSheetRight/SideSheetRight";
+import { ProductOverview, EnhancedCartForm } from "./ProductOverview";
+import {
+    render,
+    fireEvent,
+    waitForElementToBeRemoved,
+    waitFor,
+    waitForElement,
+} from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+describe("Product Overview", () => {
+    const testDetail = {
+        name: "Arduino",
+        type: "red",
+        total: 30,
+        available: 29,
+        img: "https://www.filterforge.com/more/help/images/size200.jpg",
+        category: ["MCU", "FPGA"],
+        manufacturer: "Canakit",
+        model_num: "Model 3B+",
+        datasheet: "link",
+        notes: ["- For micropython ask for image", "- randomnerdtutorials.com"],
+        constraints: ["- Max 1 of this item", "- Max 3 microcontroller labelled red"],
+        quantity: 3,
+    };
+
+    test("render", () => {
+        const addCartMock = jest.fn();
+
+        const { asFragment } = render(
+            <SideSheetRight title="Product Overview" isVisible={true}>
+                <ProductOverview detail={testDetail} addCartFunction={addCartMock} />
+            </SideSheetRight>
+        );
+
+        expect(asFragment()).toMatchSnapshot();
```

Comment:
The snapshot this produced is just an empty div, because of how drawers work (the same will happen for other pop-up components like Dialogs).

I would stick to testing things concretely, and try to avoid snapshot testing if you can. Think of the things that the user would be looking for when using your component (eg all the information is there, buttons work, etc), and test those things explicitly.